tests/utils/test_bits.py

```diff
@@ -0,0 +1,107 @@
+from math import isinf, isnan
+from random import randrange
+
+import pytest
+
+from xdsl.utils.bits import (
+    as_str,
+    from_f16,
+    from_f32,
+    from_f64,
+    from_i8,
+    from_i16,
+    from_i32,
+    from_i64,
+    from_u8,
+    from_u16,
+    from_u32,
+    from_u64,
+    to_f16,
+    to_f32,
+    to_f64,
+    to_i8,
+    to_i16,
+    to_i32,
+    to_i64,
+    to_u8,
+    to_u16,
+    to_u32,
+    to_u64,
+)
+
+
+@pytest.mark.parametrize(
+    "i",
+    (randrange(1 << 8) for _ in range(256)),
+)
+def test_8bits(i: int):
+    b = from_u8(i)
+    assert as_str(b).endswith(bin(int(b.hex(), 16)).removeprefix("0b"))
```

Comment:
What's the purpose of this line? Couldn't it just be `as_str(b)[:-8] == bin(i)[:-8]`? Why the hex to int to bin?

Reply:
Good question I see what you intend to do. Minor note is that `as_str` always has fixed length (as indicated by the bytes provided) but `bin` has dynamic length (i.e. trailing zeros are omitted) and also leading `0b` prefix.